golang/cosmos/x/swingset/types/store.go

```diff
@@ -0,0 +1,155 @@
+package types
+
+import (
+	"bytes"
+	"io"
+
+	prefixstore "github.com/cosmos/cosmos-sdk/store/prefix"
+	storetypes "github.com/cosmos/cosmos-sdk/store/types"
+)
+
+var (
+	// SwingStoreValueSuffix is the null byte appended to all SwingStore values
+	// before writing them in the underlying KVStore. This allows accepting nil
+	// values, which are expected of the JS swing-store but disallowed by a
+	// normal KVStore.
+	SwingStoreValueSuffix = []byte{0}
+)
+
+var _ storetypes.KVStore = SwingStore{}
+
+// SwingStore wraps a KVStore, possibly prefixed, allowing nil values.
+// It does so by appending a null byte to all values before writing them in the
+// parent store, and stripping a trailing byte when reading values from the
+// parent store.
+type SwingStore struct {
+	parent storetypes.KVStore
+}
```

Comment:
I find this "parent" framing to be slightly misleading.
```suggestion
// SwingStore wraps a KVStore to allow nil values by transparently suffixing
// all values with a NUL byte in the wrapped store (i.e., a nil value is stored
// as `[]byte{0}` and a value of 'A' is stored as `[]byte{'A', 0}`).
type SwingStore struct {
	baseStore storetypes.KVStore
}
```